lib/device_detector/device.rb

```diff
@@ -0,0 +1,54 @@
+class DeviceDetector
+  class Device < Parser
+
+    def known?
+      regex_meta.any?
+    end
+
+    def name
+      ModelExtractor.new(user_agent, regex_meta).call
+    end
+
+    def device_type
+      regex_meta['device']
+    end
+
+    private
+
+    def filenames
+      [
+        'devices/mobiles.yml'
+      ]
+    end
+
+    # Overwrite the default parser handling logic
+    def self.regexes_for(filepaths)
+      @regexes ||=
+        begin
+          regexes = YAML.load(filepaths.map { |filepath| File.read(filepath) }.join)
+
+          recursive_regex_parser(regexes)
+        end
+    end
+
+    def self.recursive_regex_parser(nested_regex)
+      nested_regex.map { |base, nest|
+
+        if !nest.nil? && nest.include?('models')
+          recursive_regex_parser nest['models']
+        else
+          case base.class.to_s
+            when 'Hash'
+              base['regex'] = Regexp.new base['regex']
+              base
+            else
```

Comment:
Probably there is no need for type checking in here, but what if `base` is something unexpected? Maybe it make the code more solid if there was an explicit check, and fail if input is unacceptable:

```ruby
case base
  when Hash
    base['regex'] = Regexp.new base['regex']
    base
  when String # pseudo
    nest['regex'] = Regexp.new nest['regex']
    nest
  else
    fail('Base cannot be Foo')
end
```